tensorflow/compiler/mlir/tosa/tests/tf-to-tosa-pipeline.mlir

```diff
@@ -764,22 +771,64 @@ func @test_depth_to_space(%arg0: tensor<1x32x32x8xf32>) -> tensor<1x64x64x2xf32>
 
 // -----
 
+// CHECK-LABEL: test_left_shift
+// CHECK: tosa.logical_left_shift
+func @test_left_shift(%arg0: tensor<4x4xi32>, %arg1: tensor<1x1xi32>) -> tensor<4x4xi32> {
+  %0 = "tf.LeftShift"(%arg0, %arg1) : (tensor<4x4xi32>, tensor<1x1xi32>) -> tensor<4x4xi32>
+  return %0 : tensor<4x4xi32>
+}
+
+// -----
+
+// CHECK-LABEL: test_right_shift
+// CHECK: tosa.arithmetic_right_shift
+func @test_right_shift(%arg0: tensor<4x4xi32>, %arg1: tensor<1x1xi32>) -> tensor<4x4xi32> {
+  %0 = "tf.RightShift"(%arg0, %arg1) : (tensor<4x4xi32>, tensor<1x1xi32>) -> tensor<4x4xi32>
+  return %0 : tensor<4x4xi32>
+}
+
+// -----
+
+// CHECK-LABEL: test_one_hot
+// CHECK: tosa.const
+// CHECK: tosa.reshape
+// CHECK: tosa.tile
+// CHECK: tosa.reshape
+// CHECK: tosa.tile
+// CHECK: tosa.reshape
+// CHECK: tosa.scatter
+// CHECK: tosa.reshape
+// CHECK: tosa.transpose
+// CHECK: tosa.reshape
+func @test_one_hot(%arg0: tensor<4x4xi32>, %arg1: tensor<f32>, %arg2: tensor<f32>) -> tensor<4x4x2xf32> {
```

Comment:
General note on tests (and specific note on this): These are non trivial lowerings and we should have more explicit tests. The tests do not need to verify every detail of the patterns but should minimally capture SSA values and constants. Getting the testing granularity right is a bit of an art, so if you want help on this, we can bust some examples out to a doc or gist or something and work on it together if that is helpful (been meaning to ask for test upgrades for some time and realize there may be some education here).

Reply:
Ah so we can check the SSA values as well. Didn't realize we can do that.
I'd assume all we need to do is append that to "// CHECK: tosa.XXX", for example like "// CHECK: tosa.XXX(%0)"?
or it's not that simple as I expect?

Reply:
You can see similar examples in the tensorflow to hlo lowering tests:

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/mlir/xla/tests/legalize-tf.mlir#L64

The problem with not validating the large sequences is incorrect constants or changing which values are passed to which ops.

Reply:
Nit: You can drop all the shape information in the line to make things more succinct.

I.e. The following line:

// CHECK: %[[VAR0:.*]] = "tosa.const"() {value = dense<[0, 2, 1]> : tensor<3xi32>} : () -> tensor<3xi32>

Can be simplified to

// CHECK: %[[VAR0:.*]] = "tosa.const"() {value = dense<[0, 2, 1]> : tensor<3xi32>}

Generally we want to make sure check tests are mostly a sequence of correct values.